e2e/k8s_test.go

```diff
@@ -29,3 +100,59 @@ func TestSecretsProvidedK8s(t *testing.T) {
 
 	testenv.Test(t, f.Feature())
 }
+
+func TestVariablesWithBase64DecodingSecretProvidedK8s(t *testing.T) {
+	f := features.New("variables with base64 decoding secret provided").
+		// Replaces TEST_ID_1.6_providing_variables_with_base64_decoding_successfully
+		Assess("variables with base64 decoding secret set correctly in pod", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			var stdout, stderr bytes.Buffer
+			command := []string{"printenv", "|", "grep", "VARIABLE_WITH_BASE64_SECRET"}
+
+			RunCommandInSecretsProviderPod(cfg.Client(), command, &stdout, &stderr)
+
+			assert.Contains(t, stdout.String(), "secret-value")
+
+			return ctx
+		})
+
+	testenv.Test(t, f.Feature())
+}
+
+func TestLargeDecodedVariablesSecretProvidedK8s(t *testing.T) {
+	f := features.New("large decoded variables secret provided").
+		// Replaces TEST_ID_1.7_providing_large_decoded_variable_successfully
+		Assess("large decoded variable secret set correctly in pod", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// generate random string (> 65k characters) and base64 encode it
+			str := make([]byte, 65001)
+
+			_, err := rand.Read(str)
+			if err != nil {
+				fmt.Errorf("error generating random string: %s", err)
+			}
+
+			encodedStr := base64.StdEncoding.EncodeToString(str)
+
+			// set encoded value in conjur and reload template
+			err = SetConjurSecret(cfg.Client(), "secrets/encoded", encodedStr)
+			if err != nil {
+				fmt.Errorf("error setting conjur secret: %s", err)
+			}
+
+			err = ReloadWithTemplate(cfg.Client(), K8sTemplate)
```

Comment:
It seems like this method isn't redeploying the pod as expected so the test fails, but it should be fixed in my CI PR #540 

Will go ahead and approve since everything else looks good

Reply:
Awesome. Thank you!